src/native/mqtt_connection.c

```diff
@@ -101,30 +101,33 @@ static struct mqtt_jni_async_callback *mqtt_jni_async_callback_new(
     struct mqtt_jni_connection *connection,
     jobject async_callback) {
 
-    JNIEnv *env = aws_jni_get_thread_env(connection->jvm);
+    /********** JNI ENV ACQUIRE **********/
+    JNIEnv *env = aws_jni_acquire_thread_env(connection->jvm);
```

Comment:
`mqtt_jni_async_callback_new()` is a helper function, not a callback, we could just pass `env` in as an argument